contracts/MechMarketplace.sol

```diff
@@ -194,21 +169,26 @@ contract MechMarketplace is IErrorsMarketplace {
             revert ZeroValue();
         }
 
+        // TODO What to do if there is rateDIff leftovers? Keep as a fee or record into sender's map?
+        uint256 rateDiff;
+        if (actualDeliveryRate > deliveryRate) {
+            rateDiff = actualDeliveryRate - deliveryRate;
+        }
+
         // TODO what if fee is zero just because the delivery rate is in the order of 1..10_000?
         // Calculate mech payment and marketplace fee
-        marketplaceFee = (deliveryRate * fee) / 10_000;
-        mechPayment = deliveryRate - marketplaceFee;
+        marketplaceFee = (actualDeliveryRate * fee) / 10_000;
+        mechPayment = actualDeliveryRate - marketplaceFee;
 
         // Check for zero value, although this must never happen
         if (mechPayment == 0) {
             revert ZeroValue();
         }
 
-        // Record payment into mech balance
-        mapMechBalances[mech] += mechPayment;
-
-        // Record collected fee
-        collectedFees += marketplaceFee;
+        // Record mech payment and marketplace fee into corresponding escrow balances
+        IMech.MechType mechType = IMech(mech).mechType();
+        address escrow = mapMechTypeEscrows[mechType];
+        IEscrow(escrow).adjustBalances(mech, mechPayment, marketplaceFee);
```

Comment:
This logic should probably also go to the escrow, where we just pass the MM current fee, since balance adjustment also happens on that side.